Source/TableEditingController.swift

```diff
@@ -34,7 +34,7 @@ public struct TableEditingController {
 
      - returns: `true` if the specified row is editable, `false` otherwise.
      */
-    public typealias CanEditRowConfig = (_ tableView: UITableView, _ indexPath: IndexPath) -> Bool
+    public typealias CanEditRowConfig = (_ tableView: UITableView, _ indexPath: IndexPath, _ dataSource: inout DataSource) -> Bool
```

Comment:
I was thinking for this closure if we could pass the `Item` instead of `dataSource` as I have recommended in #92  since I believe that it is quite unlikely the need to mutate the `dataSource` at this point.  🤔
What do you think?